extensions/proposals/KHR_parallel_shader_compile/extension.xml

```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- vi:set sw=2 ts=4: -->
+<?xml-stylesheet href="../../extension.xsl" type="text/xsl"?>
+<proposal href="proposals/KHR_parallel_shader_compile/">
+  <name>KHR_parallel_shader_compile</name>
+
+  <contact> <a href="https://www.khronos.org/webgl/public-mailing-list/">WebGL
+  working group</a> (public_webgl 'at' khronos.org) </contact>
+
+  <contributors>
+    <contributor>Members of the WebGL working group</contributor>
+  </contributors>
+
+  <number>XYZ</number>
+
+  <depends>
+    <api version="1.0"/>
+  </depends>
+
+  <overview>
+    <mirrors href="https://www.khronos.org/registry/OpenGL/extensions/KHR/KHR_parallel_shader_compile.txt"
+             name="KHR_parallel_shader_compile">
+    </mirrors>
+
+    <features>
+      <feature>
+        Shader compilation and program linking may be performed in a separate CPU thread.
+      </feature>
+    </features>
+  </overview>
+
+  <idl xml:space="preserve">
+    [NoInterfaceObject]
+    interface KHR_parallel_shader_compile {
+      const GLenum MAX_SHADER_COMPILER_THREADS_KHR      = 0x91B0;
+      const GLenum COMPLETION_STATUS_KHR                = 0x91B1;
+	
+      void maxShaderCompilerThreadsKHR(GLuint count);
+    };
+  </idl>
+
+  <samplecode xml:space="preserve">
+    <pre>
+    var canvas = document.createElement("canvas");
+    var gl = canvas.getContext("webgl");
+    var ext = gl.getExtension('KHR_parallel_shader_compile');
+    // Don't do this unless you really want to override the implementation-specific maximum.
+    ext.maxShaderCompilerThreadsKHR(3);
+
+    var vSource = "attribute vec2 position; void main() { gl_Position = vec4(position, 0, 1); }";
+    var fSource = "precision mediump float; void main() { gl_FragColor = vec4(1,0,0,1); }";
+
+    var vShader = gl.createShader(gl.VERTEX_SHADER);
+    gl.shaderSource(vShader, vSource);
+    gl.compileShader(vShader);
+
+    var fShader = gl.createShader(gl.FRAGMENT_SHADER);
+    gl.shaderSource(fShader, fSource);
+    gl.compileShader(fShader);
+
+    var program = gl.createProgram();
+    gl.attachShader(program, vShader);
+    gl.attachShader(program, fShader);
+    gl.linkProgram(program);
+
+    function checkCompletion() {
+      if (gl.getProgramParameter(program, ext.COMPLETION_STATUS_KHR) == true) {
```

Comment:
Let's make this code work both with and without the extension. Please start with "if (ext) { ... }" and have the fallback path (synchronous linking) do the error checks too.

Reply:
Done.